File: ruby/triangle/triangle.rb
class TriangleError < StandardError
end

class Triangle
  attr_reader :a, :b, :c

  def initialize(a,b,c)
    @a = [a,b,c].min
    @c = [a,b,c].max
    @b = a + b + c - @a - @c
    validate
  end

  def kind
    return :equilateral if a == c
    return :isosceles if a == b || b == c
    :scalene
  end

  private
  def validate
    raise TriangleError if a <= 0
    raise TriangleError if a + b <= c
  end
end

File: javascript/triangle/triangle.js
var Triangle = function(a,b,c) {
  var triangle = [a,b,c].sort(function(i,j) { return i - j; });
  var a = triangle[0];
  var b = triangle[1];
  var c = triangle[2];

  this.kind = function() {
    if (a <= 0) throw 'Triangles must have a size'
    if (a + b <= c) throw 'Violates Triangle inequality'

    if (a == c) { return "equilateral"; }
    if (a == b || b == c) { return 'isosceles'; }
    return 'scalene';
  }
}

module.exports = Triangle
